src/pages/Transaction/utils.tsx

```diff
@@ -178,36 +178,70 @@ function isAptEvent(event: Types.Event, transaction: Types.Transaction) {
   return aptEventChange.length > 0;
 }
 
-export function getCoinBalanceChanges(
-  transaction: Types.Transaction,
-): BalanceChange[] {
-  const accountToBalance = getBalanceMap(transaction);
-
-  const changes: Types.WriteSetChange[] =
-    "changes" in transaction ? transaction.changes : [];
+interface TransactionResponse {
+  fungible_asset_activities: Array<{
+    amount: number;
+    entry_function_id_str: string;
+    gas_fee_payer_address?: string;
+    is_frozen?: boolean;
+    asset_type: string;
+    event_index: number;
+    owner_address: string;
+    transaction_timestamp: string;
+    transaction_version: number;
+    type: string;
+    metadata: {
+      asset_type: string;
+      decimals: number;
+      symbol: string;
+    };
+  }>;
+}
 
-  Object.entries(accountToBalance).forEach(([key]) => {
-    changes.filter((change) => {
-      if ("address" in change && change.address === key) {
-        const data = getAptChangeData(change);
-        if (data !== undefined) {
-          accountToBalance[key].amountAfter = data.coin.value;
-          return change;
+export function useTransactionBalanceChanges(txn_version: string) {
+  const {loading, error, data} = useGraphqlQuery<TransactionResponse>(
+    gql`
+      query TransactionQuery($txn_version: String) {
+        fungible_asset_activities(
+          where: {transaction_version: {_eq: ${txn_version}}}
+        ) {
+          amount
+          entry_function_id_str
+          gas_fee_payer_address
+          is_frozen
+          asset_type
+          event_index
+          owner_address
+          transaction_timestamp
+          transaction_version
+          type
+          metadata {
+            asset_type
+            decimals
+            symbol
+          }
         }
       }
-    });
-  });
+    `,
+    {variables: {txn_version}},
+  );
 
-  const balanceList: BalanceChange[] = [];
-  Object.entries(accountToBalance).forEach(([key, value]) => {
-    balanceList.push({
-      address: key,
-      amount: value.amount,
-      amountAfter: value.amountAfter,
-    });
-  });
+  const balanceChanges: BalanceChange[] =
+    data?.fungible_asset_activities.map((a) => ({
+      address: a.owner_address,
+      amount: BigInt(a.amount),
+      amountAfter: a.amount.toString(),
```

Comment:
These two are the same?

Reply:
Nope! This is wrong. Thinking back, we decided to remove one column (I think). Was it the after column?

Reply:
yea i think so